src/main/resources/assets/js/application/recruiter/controllers/tdprWeekTableController.js

```diff
@@ -77,17 +70,30 @@ define(['angular', 'application/recruiter/tdprRecruiterModule', 'application/rec
 
             if (slot === undefined) {
                 // Add available slot for future changes
-                return $scope.changeSlotType(slot, slotId, date, person, AvailabilityEnum.available.name);
+                return $scope.changeSlotType(slot, slotId, date, person, AvailabilityEnum.full.name);
             } else {
-                var newTypeId = parseInt(AvailabilityEnum[slot.type].id) + 1;
-                var newType = _.findKey(AvailabilityEnum, {'id': newTypeId.toString()});
+                // Cycle through
+                // Available/maybe - full - init - maybe
 
-                return $scope.changeSlotType(slot, slotId, date, person, newType);
-            }
-        };
+                var newType = undefined;
+
+                switch(slot.type) {
```

Comment:
Perhaps it would be a good idea to move this logic of cycling through slot types to a service? It's good to keep the controller clear of business logic implementations when possible.